libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/RoadObjectsStore.kt

```diff
@@ -102,4 +111,60 @@ class RoadObjectsStore internal constructor(
     fun removeCustomRoadObject(roadObjectId: String) {
         navigator.roadObjectsStore?.removeCustomRoadObject(roadObjectId)
     }
+
+    /**
+     * Returns a list of [UpcomingRoadObject]
+     * @param distances a map of [String] roadObjectIds and [EHorizonObjectDistanceInfo]
+     */
+    fun getUpcomingRoadObjects(distances: Map<String, EHorizonObjectDistanceInfo>):
```

Comment:
Could we extract some of the logic into well-named `private` `fun`ctions? `getUpcomingRoadObjects` is quite long IMO.

Reply:
done